plugins/obs-qsv11/obs-qsv265.c

```diff
@@ -0,0 +1,1118 @@
+{
+	video_t *video = obs_encoder_video(obsqsv->encoder);
+	const struct video_output_info *voi = video_output_get_info(video);
+	update_latency(settings);
+	update_enhancements(settings);
+
+	const char *target_usage =
+		obs_data_get_string(settings, "target_usage");
+	const char *profile = obs_data_get_string(settings, "profile");
+	const char *rate_control =
+		obs_data_get_string(settings, "rate_control");
+	const char *latency = obs_data_get_string(settings, "latency");
+	int target_bitrate = (int)obs_data_get_int(settings, "bitrate");
+	int max_bitrate = (int)obs_data_get_int(settings, "max_bitrate");
+	int accuracy = (int)obs_data_get_int(settings, "accuracy");
+	int convergence = (int)obs_data_get_int(settings, "convergence");
+	int qpi = (int)obs_data_get_int(settings, "qpi");
+	int qpp = (int)obs_data_get_int(settings, "qpp");
+	int qpb = (int)obs_data_get_int(settings, "qpb");
+	int icq_quality = (int)obs_data_get_int(settings, "icq_quality");
+	int keyint_sec = (int)obs_data_get_int(settings, "keyint_sec");
+	bool cbr_override = obs_data_get_bool(settings, "cbr");
+	int bFrames = (int)obs_data_get_int(settings, "bframes");
+	bool enhancements = obs_data_get_bool(settings, "enhancements");
+
+	if (obs_data_has_user_value(settings, "bf"))
+		bFrames = (int)obs_data_get_int(settings, "bf");
+
+	enum qsv_cpu_platform plat = qsv_get_cpu_platform();
+	if (plat == QSV_CPU_PLATFORM_IVB || plat == QSV_CPU_PLATFORM_SNB)
+		bFrames = 0;
+
+	int width = (int)obs_encoder_get_width(obsqsv->encoder);
+	int height = (int)obs_encoder_get_height(obsqsv->encoder);
+	if (astrcmpi(target_usage, "quality") == 0)
+		obsqsv->params.nTargetUsage = MFX_TARGETUSAGE_BEST_QUALITY;
+	else if (astrcmpi(target_usage, "balanced") == 0)
+		obsqsv->params.nTargetUsage = MFX_TARGETUSAGE_BALANCED;
+	else if (astrcmpi(target_usage, "speed") == 0)
+		obsqsv->params.nTargetUsage = MFX_TARGETUSAGE_BEST_SPEED;
+	else if (astrcmpi(target_usage, "veryslow") == 0)
+		obsqsv->params.nTargetUsage = MFX_TARGETUSAGE_1;
+	else if (astrcmpi(target_usage, "slower") == 0)
+		obsqsv->params.nTargetUsage = MFX_TARGETUSAGE_2;
+	else if (astrcmpi(target_usage, "slow") == 0)
+		obsqsv->params.nTargetUsage = MFX_TARGETUSAGE_3;
+	else if (astrcmpi(target_usage, "medium") == 0)
+		obsqsv->params.nTargetUsage = MFX_TARGETUSAGE_4;
+	else if (astrcmpi(target_usage, "fast") == 0)
+		obsqsv->params.nTargetUsage = MFX_TARGETUSAGE_5;
+	else if (astrcmpi(target_usage, "faster") == 0)
+		obsqsv->params.nTargetUsage = MFX_TARGETUSAGE_6;
+	else if (astrcmpi(target_usage, "veryfast") == 0)
+		obsqsv->params.nTargetUsage = MFX_TARGETUSAGE_7;
+
+	if (astrcmpi(profile, "Main") == 0)
+		obsqsv->params.nCodecProfile = MFX_PROFILE_HEVC_MAIN;
+	else if (astrcmpi(profile, "Main 10") == 0)
+		obsqsv->params.nCodecProfile = MFX_PROFILE_HEVC_MAIN10;
+	else if (astrcmpi(profile, "Auto") == 0)
+		obsqsv->params.nCodecProfile = 0;
+
+	//
+	switch (voi->colorspace) {
```

Comment:
Please support SDR. Currently seeing green screen in video using NV12 & Rec. 709.

Reply:
This code only differs PQ from the HLG.
I'll check...

Reply:
Rec. 709 seems to be working now.